### DISCUSSION TITLE
Validation when registering operations

### DISCUSSION THREAD

**Discussion by alexanderilyin** [danielgtaylor/huma#622] [Q&A] [answered]
I noticed that if I use duplicated `OperationID` then there is no problem running service, for example:

```go
huma.Register(api, huma.Operation{
	OperationID: "operation1",
	Path:        "/operation/one",
}, controllers.One)

huma.Register(api, huma.Operation{
	OperationID: "operation1",
	Path:        "/operation/two",
}, controllers.Two)
```

In my case it was Copy & Paste problem. I wonder what's the right place in the library to add implementation for checking that all `OperationID` are unique. Probably controlled by a setting/option.

**Comment by wolveix** [comment, marked as answer]
Hey @alexanderilyin! We recently added the requirement that operation IDs are unique, as what you experienced was a bug :)
